### PR TITLE
Restructure the hierarchy of Index unit tests

### PR DESCRIPTION
**Pull request by xinrong-databricks**
Proposal: 
`databricks/koalas/tests/test_indexes.py` -> `databricks/koalas/tests/indexes/test_base.py`
Create `databricks/koalas/tests/indexes/test_datetime.py`

### CODE REVIEW

**Comment by ueshin** [comment]
You need to add `databricks/koalas/tests/indexes/__init__.py`.

**Review by ueshin — APPROVED** [review]
LGTM, pending tests.

File: databricks/koalas/tests/indexes/test_datetime.py [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+from distutils.version import LooseVersion
+
+import pandas as pd
+
+import databricks.koalas as ks
+
```
Comment (ueshin) [inline databricks/koalas/tests/indexes/test_datetime.py:22]: nit: don't need this empty line?

Comment (xinrong-databricks) [reply databricks/koalas/tests/indexes/test_datetime.py:21]: Removed, thanks!

**Comment by ueshin** [comment]
Thanks! merging.

File: databricks/koalas/tests/indexes/test_datetime.py (hunk 1 above)
Comment (ueshin) [reply databricks/koalas/tests/indexes/test_datetime.py:21]: Oops, sorry. I misread the change.

I meant the empty line between `import databricks.koalas as ks` and `from databricks.koalas ...`.
We need an empty line between `import pandas ...` and `import databricks.koalas ...`.

@xinrong-databricks Could you address in your next PR?

Comment (xinrong-databricks) [reply databricks/koalas/tests/indexes/test_datetime.py:21]: Sounds good!

**Pull request merged**